nixos/modules/services/mail/postfix.nix

```diff
@@ -584,6 +514,163 @@ in
+                List of domain names intended for local delivery using /etc/passwd and /etc/aliases.
+
+                ::: {.warning}
+                Do not include [virtual](https://www.postfix.org/VIRTUAL_README.html) domains in this list.
+                :::
+
+                <https://www.postfix.org/postconf.5.html#mydestination>
+              '';
+            };
+
+            myhostname = mkOption {
+              type = with types; nullOr types.str;
+              default = null;
+              example = "mail.example.com";
+              description = ''
+                The internet hostname of this mail system.
+
+                <https://www.postfix.org/postconf.5.html#myhostname>
+              '';
+            };
+
+            mynetworks = mkOption {
+              type = with types; nullOr (listOf str);
+              default = null;
+              example = [
+                "127.0.0.0/8"
+                "::1"
+              ];
+              description = ''
+                List of trusted remote SMTP clients, that are allowed to relay mail.
+
+                Leave unset to let Postfix populate this list based on the {option}`mynetworks_style` setting.
+
+                <https://www.postfix.org/postconf.5.html#mynetworks>
+              '';
+            };
+
+            mynetworks_style = mkOption {
+              type =
+                with types;
+                nullOr (enum [
+                  "host"
+                  "subnet"
+                  "class"
+                ]);
+              default = "host";
+              description = ''
+                The method used for generating the default value for {option}`mynetworks`, if that option is unset.
+
+                <https://www.postfix.org/postconf.5.html#mynetworks_style>
+              '';
+            };
+
+            recipient_delimiter = lib.mkOption {
+              type = with types; nullOr str;
+              default = null;
+              example = "+";
+              description = ''
+                Set of characters used as the delimiters for address extensions.
+
+                This allows creating different forwarding rules per extension.
+
+                <https://www.postfix.org/postconf.5.html#recipient_delimiter>
+              '';
+            };
+
+            relayhost = mkOption {
+              type = with types; nullOr (listOf str);
+              default = [ ];
+              example = [ "[relay.example.com]:587" ];
+              description = ''
+                List of hosts to use for relaying outbound mail.
+
+                ::: {.note}
+                Putting the hostname in angled brackets, e.g. `[relay.example.com]`, turns off MX and SRV lookups for the hostname.
+                :::
+
+                <https://www.postfix.org/postconf.5.html#relayhost>
+              '';
+            };
+
+            relay_domains = mkOption {
+              type = with types; nullOr (listOf str);
+              default = null;
```

Comment:
```suggestion
              default = [ ];
```

Reply:
I think null is probably a better default, to explicitly leave the decision to Postfix, as above.

Reply:
But Postfix's default is an empty list.